### PR TITLE
UI: paste payment request from clipboard

### PR DESCRIPTION
**Pull request by callebtc**
telegram-cloud-photo-size-2-5353022529796688625-x

### CODE REVIEW

File: lnbits/core/static/js/wallet.js [hunk 1]
```diff
@@ -795,6 +795,11 @@ new Vue({
         field: 'pending'
       })
       LNbits.utils.exportCSV(columns, this.payments)
+    },
+    pasteToTextArea: function () {
+      navigator.clipboard.readText().then(text => {
+        this.$refs.textArea.value = text
```
Comment (motorina0) [inline lnbits/core/static/js/wallet.js:801]: is this protected against XSS?

Comment (callebtc) [reply lnbits/core/static/js/wallet.js:801]: Yes
image

**Review by motorina0 — APPROVED** [review]
LGTM

**Review by dni — APPROVED** [review]

**Review by arcbtc — APPROVED** [review]

File: lnbits/core/static/js/wallet.js [hunk 2]
```diff
@@ -795,6 +795,11 @@ new Vue({
         field: 'pending'
       })
       LNbits.utils.exportCSV(columns, this.payments)
+    },
+    pasteToTextArea: function () {
+      navigator.clipboard.readText().then(text => {
```
Comment (michael1011) [inline lnbits/core/static/js/wallet.js:800]: Just fyi: `navigator.clipboard` is not available on sites that are either

- not localhost
- or not HTTPs

Some people might access their lnbits behind a VPN which is a secure connection but HTTP or a LAN; should be considered to implement a fallback for that scenario or hide the button


https://stackoverflow.com/a/65996386

**Pull request merged**